File: Python3/TowerOfHanoi.py
i=0
def towerOfHanoi(disks, source, auxiliary, target):
    global i
    if disks == 1:
        print("Move disk 1 from tower {} to tower {}.".format(source, target))
        i+=1
        return
 
    towerOfHanoi(disks - 1, source, target, auxiliary)
    print("Move disk {} from tower {} to tower {}.".format(disks, source, target))
    i+=1
    towerOfHanoi(disks - 1, auxiliary, source, target)
 

disks = int(input('Enter number of disks: \n'))
towerOfHanoi(disks, 'A', 'B', 'C')
print("Total number of steps required are {}".format(i))

File: C++/TowerOfHanoi.cpp
#include <bits/stdc++.h> 
using namespace std; 
  
void towerOfHanoi(int n, char source_rod, 
                    char destination, char third_rod)  
{  
    if (n == 1)  
    {  
        cout << "Move disk 1 from rod " << source_rod <<  
                            " to rod " << destination<<"\n";  
        return;  
    }  
    towerOfHanoi(n - 1, source_rod, third_rod, destination);  
    cout << "Move disk " << n << " from rod " << source_rod << 
                                " to rod " << destination <<"\n";  
    towerOfHanoi(n - 1, third_rod, destination, source_rod);  
}  
  

int main()  
{  
    int n; 
    cout<<"Enter the no. of disks ";
    cin>>n;
    cout<<"\nThe steps that are taken:  \n";

    towerOfHanoi(n, 'A', 'C', 'B'); 
  
} 